File: AspNetMvcUnleashed/0672329980_source/Chapter12/CS/MvcApplication1/MvcApplication1/Controllers/UserController.cs
using System.Web.Mvc;
using System;

namespace MvcApplication1.Controllers
{
    public class UserController : Controller
    {

        public ActionResult Index()
        {
            // Show ManagerView view to members of Manager role
            if (User.IsInRole("Manager"))
                return View("ManagerView");

            // Show JackView to Jack (and no one else)
            if (string.Equals(User.Identity.Name, "Jack", StringComparison.CurrentCultureIgnoreCase))
                return View("JackView");

            // Show AuthenticatedView to non-anonymous visitors
            if (User.Identity.IsAuthenticated)
                return View("AuthenticatedView");

            // Otherwise, redirect to LogOn action
            return RedirectToAction("LogOn", "Account");
        }

    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter12/VB/MvcApplication1/MvcApplication1/Controllers/UserController.vb
Public Class UserController
    Inherits Controller

    Public Function Index() As ActionResult
        ' Show ManagerView view to members of Manager role
        If User.IsInRole("Manager") Then
            Return View("ManagerView")
        End If

        ' Show JackView to Jack (and no one else)
        If String.Equals(User.Identity.Name, "Jack", StringComparison.CurrentCultureIgnoreCase) Then
            Return View("JackView")
        End If

        ' Show AuthenticatedView to non-anonymous visitors
        If User.Identity.IsAuthenticated Then
            Return View("AuthenticatedView")
        End If

        ' Otherwise, redirect to LogOn action
        Return RedirectToAction("LogOn", "Account")
    End Function

End Class
